LongWarOfTheChosen/Src/LW_Overhaul/Classes/XComGameState_LWOutpost.uc

```diff
@@ -1763,3 +1801,30 @@ function bool ShouldBeVisible()
 {
 	return false;
 }
+
+function UpdateRebelAbilities(XComGameState NewGameState)
+{
+	local RebelUnit Rebel;
+	local XComGameState_Unit UnitState;
+
+	foreach Rebels(Rebel)
+	{
+		UnitState = XComGameState_Unit(NewGameState.CreateStateObject(class'XComGameState_Unit', Rebel.Unit.ObjectID));
+		NewGameState.AddStateObject(UnitState);
+		// Rebel has ranked up but has no abilities, so give them abilities
+		if (Rebel.Level > 0 && UnitState.GetAWCAbilityNames().Length == 0)
```

Comment:
`GetAWCAbilityNames()` doesn't work because it has a check for `bUnlocked`, which none of the abilities has set to `true`. So this function always returns a zero-length array.

You either need to check the length of `AWCAbilities` directly, or set `bUnlocked` to true when giving each AWC perk.

Reply:
GiveAWCPerk() already sets bUnlocked to true, so I'm not sure why this doesn't work. But you're right, this is still firing when they already have abilities. Weird.

Reply:
Are you sure? Just looking at the last commit and it looks like the previous code was setting `bUnlocked` to `false` :wink:

Reply:
@Favid I will merge this, but just want to check: should `bUnlocked` be set to `true`? I'm assuming yes, in which case I'll make the change.

Reply:
Actually, ignore that. I was looking at where the ability was initialised, not where it was granted (at which point `bUnlocked` is indeed set to `true`). As with you, have no idea why I was running into the problem with `GetAWCAbilityNames()` but can confirm it's working.